plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/flexible/FlexDeployPreferencesPanelTest.java

```diff
@@ -132,11 +130,19 @@ private static IStatus getAppYamlPathValidationStatus(FlexDeployPreferencesPanel
   }
 
   private static Text findAppYamlField(Composite panel) {
-    return (Text) CompositeUtil.findControl(panel, new Predicate<Control>() {
+    Predicate<Control> predicate = new Predicate<Control>() {
       @Override
       public boolean apply(Control control) {
-        return control instanceof Text && APP_YAML_FIELD_TOOLTIP.equals(control.getToolTipText());
+        return control instanceof Text && 
+            "app.yaml Path, either Absolute or Relative to the Project".equals(
+                control.getToolTipText());
       }
-    });
+    };
+    
+    Control control = CompositeUtil.findControl(panel, predicate);
```

Comment:
I expect this whole thing can be simplified with `findControlAfterLabel()` which was introduce after this test was written.

```java
Control control = CompositeUtil.findControlAfterLabel(panel, Text.class, "app.yaml:");
```

Want to try?